src/ophyd_async/core/_signal.py

```diff
@@ -663,15 +663,15 @@ async def _wait_for_value():
                     break
 
         try:
-            status = asyncio.wait_for(_wait_for_value(), timeout)
+            status = AsyncStatus(asyncio.wait_for(_wait_for_value(), timeout))
```

Comment:
The `status` I was referring to is the one returned from `set_signal.set`. I was expecting:
```python
    status = set_signal.set(set_value, timeout=set_timeout)
    ...
            if wait_for_set_completion:
                await status
            return status
```